### PR TITLE
SteamVRTab SSOverride update

### PR DESCRIPTION
**Pull request by ykeara**
### New Features
- Added Supersampling Override toggle

#### Bug Fixes
- Fixed error preventing profiles from saving

#### Refactor
- Re-arranged code into more logical block/order
- Added comments to explain functions

### CODE REVIEW

File: src/tabcontrollers/SteamVRTabController.cpp [hunk 1]
```diff
@@ -309,6 +310,29 @@ void SteamVRTabController::setAllowSupersampleFiltering( bool value,
     }
 }
 
+bool SteamVRTabController::allowSupersampleOverride() const
+{
+	return m_allowSupersampleOverride;
+}
+
+void SteamVRTabController::setAllowSupersampleOverride(bool value,
```
Comment (username223) [inline src/tabcontrollers/SteamVRTabController.cpp]: Best practice is to pass immutable vars as const.

File: src/tabcontrollers/SteamVRTabController.h [hunk 2]
```diff
@@ -92,6 +98,7 @@ public slots:
     // void compositorSuperSamplingChanged( float value );
     void motionSmoothingChanged( bool value );
     void allowSupersampleFilteringChanged( bool value );
+	void allowSupersampleOverrideChanged( bool value );
```
Comment (username223) [inline src/tabcontrollers/SteamVRTabController.h]: Is it just me or is there no definition for this func?

But again, const.

File: src/tabcontrollers/SteamVRTabController.cpp [hunk 3]
```diff
@@ -111,79 +339,22 @@ Q_INVOKABLE QString
...
+/*SuperSample (value) functions*/
 
-        settingsUpdateCounter = 0;
-    }
-    else
-    {
-        settingsUpdateCounter++;
-    }
-}
+/*
```
Comment (username223) [inline src/tabcontrollers/SteamVRTabController.cpp]: Does this comment block enable Qt Creator tooltips/doc services? If it doesn't, it's probably not worth the time investment and it's probably going to be outdated pretty quickly.

Comment (ykeara) [reply src/tabcontrollers/SteamVRTabController.cpp]: not that I know of, perhaps a bit overzealous, but not that big of a time commitment, maybe will pare it back a bit, however definatly some things need documented

File: src/tabcontrollers/SteamVRTabController.cpp (hunk 1 above)
Comment (ykeara) [reply src/tabcontrollers/SteamVRTabController.cpp]: It's not immutable. Main loop and qml can both change

File: src/tabcontrollers/SteamVRTabController.cpp (hunk 3 above)
Comment (ykeara) [reply src/tabcontrollers/SteamVRTabController.cpp]: http://doc.qt.io/qt-5/01-qdoc-manual.html looking into will start using, will maybe do a complete pass across the whole program for 3.0.0

File: src/tabcontrollers/SteamVRTabController.cpp (hunk 1 above)
Comment (username223) [reply src/tabcontrollers/SteamVRTabController.cpp]: I was talking about `bool value` and `bool notify`. Passing them as `const bool value` would prevent them from being changed by accident as well as possibly enabling some compiler optimizations.

File: src/tabcontrollers/SteamVRTabController.cpp (hunk 3 above)
Comment (username223) [reply src/tabcontrollers/SteamVRTabController.cpp]: I'll add my thoughts to #41. We should probably hold off on implementing any non-standard documentation (anything other than a few lines explaining intent or summarizing logic) until we've agreed on a standard and then do pure comment PRs.

File: src/tabcontrollers/SteamVRTabController.cpp (hunk 1 above)
Comment (ykeara) [reply src/tabcontrollers/SteamVRTabController.cpp]: Modified all params to be const except the one that is altered.

**Pull request merged**